ChatCore/Protocols/Models/TypingStatusRepresenting.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  TypingStatusRepresenting.swift
+//  ChatCore
+//
+//  Created by Daniel Pecher on 05/11/2020.
+//  Copyright © 2020 Jan Schwarz. All rights reserved.
+//
+
+import Foundation
+
+public protocol TypingStatusRepresenting where Self: EntityIdentifiable {
+    var typingUsers: [EntityIdentifier: Bool] { get set }
```

Comment:
That was the first idea, but in that case we would have to download the conversation first, update the array and then store the new array (in case someone else is already typing, so that we don't override the array). This way we can just set an item in the `typingUsers` map without downloading anything.

Reply:
Ok, let's stick with this implementation